src/OnlineSales/Controllers/ContactsController.cs

```diff
@@ -10,15 +10,64 @@
 using OnlineSales.Data;
 using OnlineSales.DTOs;
 using OnlineSales.Entities;
+using OnlineSales.Interfaces;
 
 namespace OnlineSales.Controllers;
 
 [Authorize]
 [Route("api/[controller]")]
 public class ContactsController : BaseControllerWithImport<Contact, ContactCreateDto, ContactUpdateDto, ContactDetailsDto, ContactImportDto>
 {
-    public ContactsController(ApiDbContext dbContext, IMapper mapper, IOptions<ApiSettingsConfig> apiSettingsConfig)
+    private readonly IContactService contactService;
+
+    public ContactsController(ApiDbContext dbContext, IMapper mapper, IOptions<ApiSettingsConfig> apiSettingsConfig, IContactService contactService)
         : base(dbContext, mapper, apiSettingsConfig)
     {
+        this.contactService = contactService;
+    }
+
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    public override async Task<ActionResult<ContactDetailsDto>> Post([FromBody] ContactCreateDto value)
```

Comment:
@WASandaruwan this method must have the same set of errors as Patch method below, including HTTP 500.